### ISSUE TITLE
Animation Controller Dispose

### ISSUE THREAD

**Issue by kadirgulluoglu** [tlserver/flutter_map_location_marker#96]
**Describe the bug**
I get an error like "Unhandled Exception: AnimationController.dispose() called more than once"

**To Reproduce**
```dart
          FlutterMap(
            mapController: _mapController,
            options: MapOptions(
              initialZoom: 12,
              initialCenter: viewModel.latLang,
              onTap: (tapPosition, latLong) {
                viewModel.changeMarker(latLong);
              },
            ),
            children: [
              TileLayer(
                urlTemplate: 'https://tile.openstreetmap.org/{z}/{x}/{y}.png',
                userAgentPackageName: 'com.example.app',
              ),
              CurrentLocationLayer(
                followOnLocationUpdate: FollowOnLocationUpdate.once,
                turnOnHeadingUpdate: TurnOnHeadingUpdate.never,
                style: const LocationMarkerStyle(
                  marker: DefaultLocationMarker(
                    child: Icon(
                      Icons.navigation,
                      color: Colors.white,
                    ),
                  ),
                  markerSize: Size(40, 40),
                  markerDirection: MarkerDirection.heading,
                ),
              ),
              if (viewModel.markers != null)
                MarkerLayer(
                  rotate: true,
                  markers: [
                    viewModel.markers!,
                  ],
                ),
            ],
          ),
```

**Expected behavior**
clear console

**Smartphone (please complete the following information):**
 - Device: iphone 15 pro max

**Comment by Ax0elz** [comment]
I am seeing the same issue:
`[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: AnimationController.dispose() called more than once.
A given AnimationController cannot be disposed more than once.
The following AnimationController object was disposed multiple times:
  AnimationController#0ee7e(▶ 0.000; paused; DISPOSED)`

**Comment by tlserver** [comment]
Cannot reproduce, please provide
(1) MRE, minimal reproducible example,
(2) detailed step to reproduce and
(3) full error message.
Also use [code highlighting](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code-and-syntax-highlighting) for the code.

**Comment by brandonchaffee** [comment]
I am also receiving this error. See below:

`The following AnimationController object was disposed multiple times:
  AnimationController#93359(▶ 0.334; paused; DISPOSED)
#0      AnimationController.dispose.<anonymous closure> (package:flutter/src/animation/animation_controller.dart:791:9)
#1      AnimationController.dispose (package:flutter/src/animation/animation_controller.dart:802:6)
#2      _CurrentLocationLayerState._rotateMarker (package:flutter_map_location_marker/src/current_location_layer.dart:529:39)
#3      _CurrentLocationLayerState._subscriptHeadingStream.<anonymous closure> (package:flutter_map_location_marker/src/current_location_layer.dart:363:13)`

**Comment by madina-kamet** [comment]
sam error, iphone 11

**Comment by tlserver** [comment]
It would help if someone can provide
(1) MRE, minimal reproducible example,
(2) detailed step to reproduce and
(3) full error message.

"+1" or "same error" or similar message does not provide any information to fix.

**Comment by madina-kamet** [comment]
@tlserver 
Widget that I am using for this
```
CurrentLocationLayer(
                        positionStream: _positionStream,
                        alignPositionStream: _followCurrentLocationStreamController?.stream,
                        alignPositionOnUpdate: _followOnLocationUpdate,
                        alignDirectionStream: _turnHeadingUpStreamController?.stream,
                        alignDirectionOnUpdate: _turnOnHeadingUpdate,
                        style: LocationMarkerStyle(
                          marker: Container(
                            decoration: BoxDecoration(
                              color: Colors.blue,
                              borderRadius: BorderRadius.circular(30.0),
                              border: Border.all(style: BorderStyle.solid, color: Colors.white, width: 5.0),
                            ),
                          ),
                          markerSize: const Size(25.0, 25.0),
                        ),
                      )
```
ERROR 

```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: AnimationController.dispose() called more than once.
A given AnimationController cannot be disposed more than once.
The following AnimationController object was disposed multiple times:
  AnimationController#70dd4(▶ 0.000; paused; DISPOSED)
#0      AnimationController.dispose.<anonymous closure> (package:flutter/src/animation/animation_controller.dart:823:9)
#1      AnimationController.dispose (package:flutter/src/animation/animation_controller.dart:834:6)
#2      _CurrentLocationLayerState._rotateMarker (package:flutter_map_location_marker/src/widgets/current_location_layer.dart:614:39)
#3      _CurrentLocationLayerState._subscriptHeadingStream.<anonymous closure> (package:flutter_map_location_marker/src/widgets/current_location_layer.dart:452:13)
#4      _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
#5      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#6      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
#7      _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:123:11)
#8      _MapStream._handleData (dart:async/stream_pipe.dart:218:10)
#9      _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:153:13)
#10     _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
#11     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#12     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
#13     _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:123:11)
#14     _WhereStream._handleData (dart:async/stream_pipe.dart:195:12)
#15     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:153:13)
#16     _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
#17     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#18     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
#19     _ForwardingStreamSubscription._add (dart:async/stream_pipe.dart:123:11)
#20     _MapStream._handleData (dart:async/stream_pipe.dart:218:10)
#21     _ForwardingStreamSubscription._handleData (dart:async/stream_pipe.dart:153:13)
#22     _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
#23     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#24     _DelayedData.perform (dart:async/stream_impl.dart:515:14)
#25     _PendingEvents.handleNext (dart:async/stream_impl.dart:620:11)
#26     _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:591:7)
#27     _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
#28     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
```

**Comment by tlserver** [comment]
How can this be reproduced, or when do you receive this error? Does it occur after clicking a button or when leaving the page?

What about setting the `AnimationController` to `null` in the `dispose()` method? Could this resolve the issue?

```dart
@Override
void dispose() {
  _positionStreamSubscription.cancel();
  _headingStreamSubscription?.cancel();
  _followCurrentLocationStreamSubscription?.cancel();
  _turnHeadingUpStreamSubscription?.cancel();
  _moveMapAnimationController?.dispose();
  _moveMapAnimationController = null;
  _moveMarkerAnimationController?.dispose();
  _moveMarkerAnimationController = null;
  _rotateMapAnimationController?.dispose();
  _rotateMapAnimationController = null;
  _rotateMarkerAnimationController?.dispose();
  _rotateMarkerAnimationController = null;
  super.dispose();
}

**Comment by ibraheemalayan** [comment]
Getting the same error here.

**Comment by tlserver** [comment]
We haven't received any reports of crashes for several weeks now. I'm assuming this issue has been resolved in v8.0.4. Closing this issue.

**Comment by brandonchaffee** [comment]
Still crashing from this issue. Please reopen

**Comment by tlserver** [comment]
We haven't received any reports of crashes for several weeks now. I'm assuming this issue has been resolved in v8.0.6. Closing this issue.
